plugin/arcconf/arcconf.py

```diff
@@ -0,0 +1,919 @@
+
+
+
+  
+    @_handle_errors
+    def volume_raid_create(self, name, raid_type, disks, strip_size,
+                           flags=Client.FLAG_RSVD):
+
+        arcconf_raid_level = _lsm_raid_type_to_arcconf(raid_type)
+        arcconf_disk_ids = []
+        ctrl_num = None
+        disk_channel = ''
+        disk_device = ''
+        disk_dict = {'Channel':'0', 'Device':'1'}
+        disk_list_str = ''
+        vol_id = ''
+        lsm_vols = []
+
+        for disk in self.disks():
+            if not disk.plugin_data:
+                raise LsmError(
+                    ErrorNumber.INVALID_ARGUMENT,
+                    "Illegal input disks argument: missing plugin_data "
+                    "property")
+            (cur_ctrl_num, disk_channel, disk_device, disk_status) = disk.plugin_data.split(',')
+            if ctrl_num and cur_ctrl_num != ctrl_num:
+                raise LsmError(
+                    ErrorNumber.INVALID_ARGUMENT,
+                    "Illegal input disks argument: disks are not from the "
+                    "same controller/system.")
+
+            requested_disks = [d.name for d in disks]
+            for disk_name in requested_disks:
+                if str(disk_name) == str(disk.name):
+                    disk_channel = str(disk_channel.strip())
+                    disk_device = str(disk_device.strip())
+                    disk_dict.update({'Channel':disk_channel, 'Device': disk_device})
+                    arcconf_disk_ids.append(disk_dict.copy())
+                    disk_dict = {}
+                    ctrl_num = cur_ctrl_num
+
+        cmds = ["create", ctrl_num, "logicaldrive", "1024", arcconf_raid_level]
+        for disk_channel_device in arcconf_disk_ids:
+            cmds.append(disk_channel_device['Channel'])
+            cmds.append(disk_channel_device['Device'])
+
+        try:
+            self._arcconf_exec(cmds, COMPONENT[5], flag_convert=False, flag_force=True)
+        except ExecError:
+            # Check whether disk is free
+            requested_disk_ids = [d.id for d in disks]
+            
+            for cur_disk in self.disks():
+                if cur_disk.id in requested_disk_ids and \
+                   not ((cur_disk.status & Disk.STATUS_FREE) or (cur_disk.status & Disk.STATUS_OK)):
+                    raise LsmError(
+                        ErrorNumber.DISK_NOT_FREE,
+                        "Disk %s is not in STATUS_FREE state" % cur_disk.id)
+            
+           
+            raise
+
+         
+        # Generate pool_id from system id and array.
+        getconfig_lds_info = self._arcconf_exec(['getconfigJSON', ctrl_num ], COMPONENT[5], False)
+        getconfig_lds_info_json = getconfig_lds_info.split("\n")[1]
+        decoded_json = json.loads(getconfig_lds_info_json)
+
+        sys_id = ctrl_num
+        cnt = ctrl_num
+
+        
+        try:
+            latest_ld = len(decoded_json['Controller']['LogicalDrive']) - 1
```

Comment:
Checking the `latest_ld` here might cause problem when 2+ simultaneous call of `volume_raid_create()`.